### PR TITLE
Update Terraform code for COOL multi-account environment

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description

In this pull request I update this repo to work with our current implementation of the COOL multi-account environment.

## 💭 Motivation and Context

This code is to be deployed to the COOL.  Therefore it must work with our COOL multi-account environment.

## 🧪 Testing

I deployed these changes to production, and I verified that all the pre-commit hooks passed.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: provisionnetworking_policy.tf [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# ------------------------------------------------------------------------------
+# Create the IAM policy that allows all of the permissions necessary
+# to provision the networking layer in the Shared Services account.
+# ------------------------------------------------------------------------------
+
+data "aws_iam_policy_document" "provisionnetworking_policy_doc" {
+  # This policy needs to be more restricted, obviously
```
Comment (dav3r) [inline provisionnetworking_policy.tf]: 🎗Reminder to create an issue to go back and restrict this policy (if this PR is merged and this policy is still wide open).

**Review by dav3r — APPROVED** [review]
Looks great, just the one reminder... 🤓

**Review by mcdonnnj — APPROVED** [review]
I second @dav3r 's comment, but LGTM 👍

**Review by felddy — APPROVED** [review]
Looks good.  🎗

File: provisionnetworking_policy.tf (hunk 1 above)
Comment (jsf9k) [reply provisionnetworking_policy.tf]: See commit 5eb9f36.

**Pull request merged**